go-controller/pkg/mock/mock_lsp.go

```diff
@@ -0,0 +1,236 @@
+}
+
+// Set dhcp6_options uuid on lsp
+func (mock *MockOVNClient) LSPSetDHCPv6Options(lsp string, options string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get dhcp6_options from lsp
+func (mock *MockOVNClient) LSPGetDHCPv6Options(lsp string) (*goovn.DHCPOptions, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Set options in LSP
+func (mock *MockOVNClient) LSPSetOptions(lsp string, options map[string]string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get Options for LSP
+func (mock *MockOVNClient) LSPGetOptions(lsp string) (map[string]string, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Set dynamic addresses in LSP
+func (mock *MockOVNClient) LSPSetDynamicAddresses(lsp string, address string) (util.OVNCommandInterface, error) {
+	return &MockOVNCommand{
+		Exe:     mock,
+		op:      OpUpdate,
+		table:   LogicalSwitchPortType,
+		objName: lsp,
+		objUpdate: UpdateCache{
+			FieldType:  LogicalSwitchPortDynamicAddresses,
+			FieldValue: address,
+		},
+	}, nil
+
+}
+
+// Get dynamic addresses from LSP
+func (mock *MockOVNClient) LSPGetDynamicAddresses(lsp string) (string, error) {
+	lspRet, err := mock.LSPGet(lsp)
+	if err != nil {
+		return "", err
+	}
+	if lspRet != nil {
+		return "", fmt.Errorf("no lsp found with name: %s", lsp)
+	}
+	return lspRet.DynamicAddresses, nil
+}
+
+// Set external_ids for LSP
+func (mock *MockOVNClient) LSPSetExternalIds(lsp string, external_ids map[string]string) (util.OVNCommandInterface, error) {
+	return &MockOVNCommand{
+		Exe:     mock,
+		op:      OpUpdate,
+		table:   LogicalSwitchPortType,
+		objName: lsp,
+		objUpdate: UpdateCache{
+			FieldType:  LogicalSwitchPortExternalId,
+			FieldValue: external_ids,
+		},
+	}, nil
+}
+
+// Get external_ids from LSP
+func (mock *MockOVNClient) LSPGetExternalIds(lsp string) (map[string]string, error) {
+	lspRet, err := mock.LSPGet(lsp)
+	if err != nil {
+		return nil, err
+	}
+	if lspRet != nil {
+		return nil, fmt.Errorf("no lsp found with name: %s", lsp)
+	}
+	extIds := make(map[string]string)
+	for k, v := range lspRet.ExternalID {
+		key, keyOk := k.(string)
+		value, valueOk := v.(string)
+		if !keyOk || !valueOk {
+			continue
+		}
+		extIds[key] = value
+	}
+	return extIds, nil
+}
+
+// helper function that applies field updates for a given lsp to the mock object cache
+func (mock *MockOVNClient) updateLSPCache(lspName string, update UpdateCache, mockCache MockObjectCacheByName) error {
+	var err error = nil
+	if entry, ok := mockCache[lspName]; !ok {
+		err = fmt.Errorf("error updating LSP with name %s, LSP doesn't exist", lspName)
+	} else {
+		if lsp, ok := entry.(*goovn.LogicalSwitchPort); !ok {
+			err = fmt.Errorf("type assertion failed for LSP cache entry")
```

Comment:
You can just `panic()` here - this would be a coding error internal to the library.

Reply:
done.